packages/sit-onyx/src/i18n/index.ts

```diff
@@ -87,11 +79,12 @@ const createI18n = (options: ProvideI18nOptions = {}) => {
    * Current locale.
    * @default "en-US"
    */
-  const locale = computed(() => unref(options?.locale) ?? "en-US");
+  const locale = computed(() => toValue(options?.locale) ?? "en-US");
 
-  // If the user provided a custom `t` function it should be used instead of the default one
+  // If the user provided a custom `t` function, it is used instead of the default.
+  // Then we also skip the loading and applying of messages.
   if (options.t) {
-    return { t: options.t, locale };
+    return { t: computed(() => toValue(options.t!)), locale } as const;
```

Comment:
I am not sure if `!` is safe here since the if check only works initially but the computed function might be re-run later due to reactivity